sdk/openai/Azure.AI.OpenAI/tests/ChatCompletionsTests.cs

```diff
@@ -81,6 +81,65 @@ public async Task ChatCompletionsContentFilterCategories(Service serviceTarget)
             AssertExpectedContentFilterResponseResults(firstChoice.ContentFilterResults, serviceTarget);
         }
 
+        [RecordedTest]
+        [TestCase(Service.Azure)]
+        [TestCase(Service.NonAzure)]
+        public async Task ChatCompletionsLogprobs(Service serviceTarget)
+        {
+            OpenAIClient client = GetTestClient(serviceTarget);
+            // logprobs seems to only currently be live on gpt-35-turbo for azure openai
+            string deploymentOrModelName = GetDeploymentOrModelName(serviceTarget, Scenario.Completions);
+            ChatCompletionsOptions requestOptions = new()
+            {
+                DeploymentName = deploymentOrModelName,
+                Messages =
+                {
+                    new ChatRequestUserMessage("Say this is a test!"),
+                },
+                EnableLogProbabilities = true
+            };
+            Response<ChatCompletions> response = await client.GetChatCompletionsAsync(requestOptions);
+
+            Assert.That(response, Is.Not.Null);
+            Assert.That(response.Value, Is.Not.Null);
+            Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+            Assert.That(response.Value.Choices[0].LogProbabilityInfo?.TokenLogProbabilityResults, Is.Not.Null.Or.Empty);
+        }
+
+        [RecordedTest]
+        [TestCase(Service.Azure)]
+        [TestCase(Service.NonAzure)]
+        public async Task ChatCompletionsTopLogprobs(Service serviceTarget)
+        {
+            OpenAIClient client = GetTestClient(serviceTarget);
+            // logprobs seems to only currently be live on gpt-35-turbo for azure openai
+            string deploymentOrModelName = GetDeploymentOrModelName(serviceTarget, Scenario.Completions);
+            int topLogprobs = 3;
+            ChatCompletionsOptions requestOptions = new()
+            {
+                DeploymentName = deploymentOrModelName,
+                Messages =
+                {
+                    new ChatRequestUserMessage("Say this is a test!"),
+                },
+                EnableLogProbabilities = true,
+                LogProbabilitiesPerToken = topLogprobs
+            };
+            Response<ChatCompletions> response = await client.GetChatCompletionsAsync(requestOptions);
+
+            Assert.That(response, Is.Not.Null);
+            Assert.That(response.Value, Is.Not.Null);
+            Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+            Assert.That(response.Value.Choices[0].LogProbabilityInfo?.TokenLogProbabilityResults, Is.Not.Null.Or.Empty);
+            var probResults = response.Value.Choices[0].LogProbabilityInfo.TokenLogProbabilityResults;
+
+            foreach (ChatTokenLogProbabilityResult i in probResults)
```

Comment:
It's really not super important, but I would remmendend to use another name (i.e. `probResult`) instead of `i` which makes me think it's a number. 

```suggestion
            foreach (var probResult in probResults)
```

Reply:
Resolved by https://github.com/ninghu/azure-sdk-for-net/pull/15/commits/06519dce9d32c729a1044fa07e6f7765551ab05a